modules/experimental-layers/src/contour-layer/contour-utils.js

```diff
@@ -0,0 +1,44 @@
+import MarchingSquares from './marching-squares';
+import assert from 'assert';
+
+export function generateContours({
+  thresholds,
+  colors,
+  cellWeights,
+  gridSize,
+  gridOrigin,
+  cellSize
+}) {
+  const contourSegments = [];
+
+  thresholds.forEach((threshold, index) => {
+    const numCols = gridSize[0];
+    for (let cellIndex = 0; cellIndex < gridSize[0] * (gridSize[1] - 1); cellIndex++) {
+      if (cellIndex === 0 || (cellIndex + 1) % numCols !== 0) {
+        const code = MarchingSquares.getCode({
```

Comment:
Maybe some brief comments here before each call, to guide the reader who is not so familiar with marching squares?